src/Paket.Core/Versioning/Requirements.fs

```diff
@@ -791,25 +791,30 @@ type PackageRequirement =
     member this.Depth = this.Graph.Length
 
     static member Compare(x,y,startWithPackage:PackageFilter option,boostX,boostY) =
-        if x = y then 0 else
-        seq {
-            yield compare
-                (not x.VersionRequirement.Range.IsGlobalOverride,x.Depth)
-                (not y.VersionRequirement.Range.IsGlobalOverride,y.Depth)
-            yield match startWithPackage with
+        if obj.ReferenceEquals(x, y) then 0 else
```

Comment:
yes and yes.

ref-equals is just a single pointer comparison, the custom equality is one (potentially de-virtualized) virtual call, one type-test (https://github.com/0x53A/Paket/blob/932400899c3927c823cc9b869de545de71c650f2/src/Paket.Core/Versioning/Requirements.fs#L771) and up to 6 member comparisons. And if any of these members are non-equal, then they are just compared again in Compare itself.

Afaik profiling showed a single-digit-% performance improvement, but I want to rerun these tests anyway.